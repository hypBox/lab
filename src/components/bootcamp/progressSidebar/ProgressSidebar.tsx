import React from "react"
import { createStyles, makeStyles } from "../../material/styles"
import { List } from "../../material/core"
import { IMilestone, IBootcamp, ITask } from "../../../types"
import Milestone from "./Milestone"

const useStyles = makeStyles((theme) =>
  createStyles({
    root: {
      width: "100%",
      maxWidth: "100%",
      backgroundColor: theme.palette.background.default,
    },
  })
)

export interface ProgressSidebarProps {
  bootcamp: IBootcamp
  onMilestoneClick: CallableFunction
  onTaskClick: CallableFunction
  selectedTask: ITask
  selectedMilestone: IMilestone
}

const ProgressSidebar: React.FunctionComponent<ProgressSidebarProps> = ({
  bootcamp,
  onMilestoneClick,
  onTaskClick,
  selectedMilestone,
  selectedTask,
}) => {
  const classes = useStyles()
  const { milestones } = bootcamp

  return (
    <List
      component="nav"
      aria-labelledby="nested-list-subheader"
      className={classes.root}
    >
      {milestones.map((milestone: IMilestone) => (
        <Milestone
          key={milestone.id}
          milestone={milestone}
          onMilestoneClick={onMilestoneClick}
          onTaskClick={onTaskClick}
          selectedTask={selectedTask}
          selectedMilestone={selectedMilestone}
        />
      ))}
    </List>
  )
}

export default ProgressSidebar
