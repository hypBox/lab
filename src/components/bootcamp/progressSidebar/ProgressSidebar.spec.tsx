import React from "react"
import renderer from "react-test-renderer"

import ProgressSidebar, { ProgressSidebarProps } from "./ProgressSidebar"

describe("ProgressSidbar", () => {
  const task = {
    id: "t1",
    title: "Bootcamp Overview",
    type: "video",
    length: "5m 14s",
    progress: 100,
  }

  const milestone = {
    id: "m1",
    title: "Overview",
    progress: 100,
    tasks: [task],
  }
  const bootcamp = {
    id: "b1",
    title: "Python For Beginners",
    progress: 12.4,
    currentTask: {
      milestoneId: "m2",
      taskId: "t4",
    },
    milestones: [milestone],
  }
  const props: ProgressSidebarProps = {
    bootcamp,
    onMilestoneClick: jest.fn(),
    onTaskClick: jest.fn(),
    selectedMilestone: milestone,
    selectedTask: task,
  }

  it("renders as expected", () => {
    const tree = renderer.create(<ProgressSidebar {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
