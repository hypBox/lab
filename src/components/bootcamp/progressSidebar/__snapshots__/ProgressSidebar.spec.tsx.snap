// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressSidbar renders as expected 1`] = `
<nav
  aria-labelledby="nested-list-subheader"
  className="MuiList-root makeStyles-root-1 MuiList-padding"
>
  <div
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemIcon-root"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-check fa-w-16 "
        data-icon="check"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-displayBlock"
      >
        Overview
      </span>
    </div>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-chevron-down fa-w-14 "
      color="error"
      data-icon="chevron-down"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
  <div
    className="MuiCollapse-container MuiCollapse-entered"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          className="MuiList-root MuiList-dense"
        >
          <li
            className="MuiListItem-container"
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root makeStyles-nested-63 MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction Mui-selected"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-check fa-w-16 "
                  data-icon="check"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </div>
              <div
                className="MuiListItemText-root MuiListItemText-dense"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-displayBlock"
                >
                  Bootcamp Overview
                </span>
              </div>
            </div>
            <div
              className="MuiListItemSecondaryAction-root"
            >
              <span
                className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextPrimary"
              >
                5m 14s
              </span>
            </div>
          </li>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
